app/controllers/profiles_controller.rb

```diff
@@ -10,8 +10,12 @@ def edit
 
   def update
     if @user.update_attributes user_params
-      flash[:notice] = I18n.t('flash.notice.profile_updated')
-      set_mail_changes if changed_email?
+      flash[:notice] = if changed_email?
+        I18n.t('flash.notice.profile_email_updated')
+      else
+        I18n.t('flash.notice.profile_updated')
```

Comment:
I would use only `flash.notice.profile_updated`. Email is part of the user profile so I don't see much sense in using different flash messages.